src/database-controller/src/initializer/index.js

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+require('module-alias/register');
+require('dotenv').config();
+const DatabaseModel = require('openpaidbsdk');
+const logger = require('@dbc/core/logger');
+const { paiVersion, paiCommitVersion } = require('@dbc/package.json');
+const k8s = require('@dbc/core/k8s');
+const { Snapshot } = require('@dbc/core/framework');
+
+async function updateFromNoDatabaseVersion(databaseModel) {
+  // update from 1.0.0 < version < v1.2.0
+  await databaseModel.synchronizeSchema();
+  // transfer old frameworks from api server to db
+  const frameworks = (await k8s.listFramework()).body.items;
+  for (const framework of frameworks) {
+    const snapshot = new Snapshot(framework);
+    logger.info(`Transferring framework ${snapshot.getName()} to database.`);
+    const record = snapshot.getRecordForLegacyTransfer();
+    record.requestSynced = true;
+    await databaseModel.Framework.upsert(record);
+  }
```

Comment:
Can recover different frameworks in concurrent way?

Reply:
Yes, done.